### PR TITLE
added assistant chat

### PR DESCRIPTION
**Pull request by kaihaozhao**
Added chat function, delete/update code ability has not been added yet.

Screen Recording 2024-01-25 at 8 04 45 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,26 @@
   ],
   "main": "./out/extension.js",
   "contributes": {
+    "viewsContainers": {
+      "activitybar": [
+          {
+            "id": "assistantChat",
+            "title": "Assistant Chat",
```
Comment (codemakerai-dev) [inline package.json]: Can we brand this with our name and just call it "CodeMaker AI".

File: media/main.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            const language = hljs.getLanguage(lang) ? lang : 'plaintext';
+            return hljs.highlight(code, { language }).value;
+        }
+    });
+    document.querySelectorAll('.message pre code').forEach((block) => {
+        hljs.highlightBlock(block);
+    });
+}
+
+function addCopyButtonToCodeBlocks() {
```
Comment (codemakerai-dev) [inline media/main.js:119]: This is pretty neat.

I was thinking in the long run we could also have "Apply" button. We know to which part of the original code the response corresponds to, so we could merge it back with a single button press.

File: package.json [hunk 3]
```diff
@@ -54,42 +74,48 @@
           "default": false,
           "order": 2
         },
+        "codemaker.enableAssistant": {
+          "description": "Enable assistant.",
+          "type": "boolean",
+          "default": false,
```
Comment (codemakerai-dev) [inline package.json]: Let's enable it by default, there is no user downside to it.

File: package.json [hunk 4]
```diff
@@ -54,42 +74,48 @@
...
+          "order": 3
+        },
```
Comment (codemakerai-dev) [inline package.json:82]: I added one extra setting to InteliJ, which is to control the assistant code completions and opt out of them, the settings will result in disabling the source context contextual operations reverting to a regular chat experience.

I am thinking whether that is needed or not, but if the user experience would become degraded by too aggressive attempts to match the code user could always op out of them.

**Comment by codemakerai-dev** [comment]
Looks great. Thanks for making the change.

Screenshot 2024-01-25 at 8 53 45 PM

File: package.json (hunk 1 above)
Comment (kaihaozhao) [reply package.json]: Will fix

File: media/main.js (hunk 2 above)
Comment (kaihaozhao) [reply media/main.js:119]: Exactly, I am thinking of this feature as well, even more we can add, like after click "Apply", we modify the source code with a `diff view`, and pop up two "Accept", "Discard" button in the editor.

File: package.json (hunk 4 above)
Comment (kaihaozhao) [reply package.json:82]: Will check. by regular chat experience do you mean AssistantChatCompletion? Which doesn't include any code, only message?

**Pull request merged**